routemaster/models/queue.rb

```diff
@@ -1,130 +1,189 @@
 require 'routemaster/models'
-require 'routemaster/models/message'
-require 'routemaster/mixins/log'
 require 'routemaster/mixins/redis'
-require 'routemaster/services/codec'
+require 'routemaster/mixins/log'
+require 'routemaster/mixins/log_exception'
+require 'routemaster/models/job'
 
 module Routemaster
   module Models
-    # Abstraction for a queue.
-    # Takes a Subscriber and allows to push Messages in and pull them out in
-    # order.
+    # A collection of `Job`s, scheduled or queued, with deduplication and safe
+    # mutation (push/pop).
     class Queue
-      include Mixins::Log
       include Mixins::Redis
+      include Mixins::Log
+      include Mixins::LogException
 
-      attr_reader :subscriber
+      attr_reader :name
 
-      def initialize(subscriber)
-        @subscriber = subscriber
+      class Retry < StandardError
+        attr_reader :delay
+        def initialize(delay)
+          @delay = delay
+        end
       end
 
-      def pop
-        uid, payload = _redis_lua_run(
-          'pop',
-          keys: [_new_uuids_key, _pending_uuids_key, _payloads_key],
-          argv: [Routemaster.now])
 
-        return if uid.nil?
+      def initialize(name:)
+        @name = name
+      end
 
-        if payload.nil?
-          _log.error { "missing payload for message #{uid} in queue #{@subscriber.name}" }
-          return
+
+      # Returns a list of all queued jobs.
+      # Only use this for testing.
+      def dump
```

Comment:
`dump` seems a weird name here; that's normally for serializing to an output format. `jobs` seems more like what it's actually doing?